sdc11073/wsdiscovery.py

```diff
@@ -992,6 +1011,7 @@ def schedule_stop(self):
         self._quitRecvEvent.set()
         self._quitSendEvent.set()
         self._recvThread.join(1)
+        self._qread_thread.join(1)

```

Comment:
Needs to be added to the join method below. Also, add timeouts to the join method below